Find a functionally equivalent code:android.preference.Preference.onBindView:COMMENT
Method Modifier: protected   
Comment:/**
 * Binds the created View to the data for this Preference.
 * <p>
 * This is a good place to grab references to custom Views in the layout and
 * set properties on them.
 * <p>
 * Make sure to call through to the superclass's implementation.
 *
 * @param view The View that shows this Preference.
 * @see #onCreateView(ViewGroup)
 */

@@ -19,29 +19,29 @@
             summaryView.setVisibility(View.GONE);
         }
     }
-    ImageView imageView = (ImageView) view.findViewById(com.android.internal.R.id.icon);
+    final ImageView imageView = (ImageView) view.findViewById(com.android.internal.R.id.icon);
     if (imageView != null) {
         if (mIconResId != 0 || mIcon != null) {
             if (mIcon == null) {
-                mIcon = getContext().getResources().getDrawable(mIconResId);
+                mIcon = getContext().getDrawable(mIconResId);
             }
             if (mIcon != null) {
                 imageView.setImageDrawable(mIcon);
@@ -31,36 +31,40 @@
         }
         imageView.setVisibility(mIcon != null ? View.VISIBLE : View.GONE);
     }
+    final View imageFrame = view.findViewById(com.android.internal.R.id.icon_frame);
+    if (imageFrame != null) {
+        imageFrame.setVisibility(mIcon != null ? View.VISIBLE : View.GONE);
+    }
     if (mShouldDisableView) {
         setEnabledStateOnViews(view, isEnabled());
     }

